pipelines/batch/src/main/java/org/openmrs/analytics/FhirSearchUtil.java

```diff
@@ -75,23 +76,28 @@ public Bundle searchByUrl(String searchUrl, int count, SummaryEnum summaryMode)
    * @param resourceList the resource types to be processed
    * @return a Map storing the counts of each resource type
    */
-  public Map<String, Integer> searchResourceCounts(String resourceList) {
+  public Map<String, Integer> searchResourceCounts(String resourceList, String since) {
     HashSet<String> resourceTypes = new HashSet<String>(Arrays.asList(resourceList.split(",")));
     HashMap<String, Integer> hashMap = new HashMap<String, Integer>();
     for (String resourceType : resourceTypes) {
       try {
         String searchUrl = resourceType + "?";
         IGenericClient client = openmrsUtil.getSourceClient();
-        Bundle result =
+        IQuery<Bundle> query =
             client
                 .search()
                 .byUrl(searchUrl)
                 .summaryMode(SummaryEnum.COUNT)
-                .returnBundle(Bundle.class)
-                .execute();
+                .returnBundle(Bundle.class);
+        if (since != null && !since.isEmpty()) {
+          query.lastUpdated(new DateRangeParam(since, null));
```

Comment:
should we create a constant variable named `UNTIL` that we set to `null` for now? Maybe in the future we have a method to select patients between two date ranges

Reply:
Actually I need this to be open-ended and I never intend to add the support for having an end to the fetch period. That is one reason I decided to use a new `--since` option instead of re-purposing the `--activePeriod` one. I can tell you the details but the gist of it is `res.res_ver = ver.res_ver` constraint in the JDBC query (see changes in `JdbcFetchHapi.java`). If you think about it, it is not easy to implement an end to this without an extra shuffle.

Reply:
Sounds good! Yeah, we can chat about the details next time we see each other